src/System.Linq/tests/WhereTests.cs

```diff
@@ -0,0 +1,832 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class WhereTests
+    {
+        private static readonly Func<int, bool> s_truePredicate = (value) => true;
+        private static readonly Func<int, int, bool> s_truePredicateWithIndex = (value, index) => true;
+
+        private static readonly Func<int, bool> s_falsePredicate = (value) => false;
+        private static readonly Func<int, int, bool> s_falsePredicateWithIndex = (value, index) => false;
+
+        private static readonly Func<int, bool> s_nullPredicate = null;
+        private static readonly Func<int, int, bool> s_nullPredicateWithIndex = null;
+        
+        private static readonly Func<int, bool> s_evenPredicate = (value) => value % 2 == 0;
+        private static readonly Func<int, bool> s_oddPredicate = (value) => value % 2 == 1;
+
+        private static readonly Func<int, int, bool> s_complexPredicateWithIndex = (value, index) => { return (value % 2 == 0 && index % 2 == 0); };
```

Comment:
IMHO, separating out the lambda like this makes it more difficult to read / understand the test, as it's now relying on functionality that's not local to the test.